### ISSUE TITLE
Consider False as equal to boolean false

### ISSUE THREAD

**Issue by trwnh** [tootsuite/mastodon#12496]
Python uses `True` and `False` for its boolean values. Passing these native values into e.g. the requests library somehow both get detected as `true` in Mastodon's API, which is confusing. Mastodon seems to coerce 0, f, and OFF (among other values) into boolean false, so it can't hurt to also do the same for False where appropriate.

**Comment by nightpool** [comment]
Mastodon's API uses JSON. if your library can't speak JSON correctly, it
won't work with mastodon's API.

**Comment by trwnh** [comment]
Well, 0, f, and OFF are not technically valid booleans in JSON, but they still work when checked for truthiness. Per the logic in #6575 "if someone was sending `exclude_replies=false` in the request I don't think they were expecting replies to still be excluded." By similar logic, sending `params={'data[alerts][follow]': False}` to the Push API would expectedly result in the follow alert being set to false, not to true.

**Comment by Tagadda** [comment]
`{"data[alerts][follow]": false}`, `{"data[alerts][follow]": "f"}`, `{"data[alerts][follow]": 0}` are valid JSON.
`false` is a Boolean, `"f"` a String and `0` a Integer.
#6575 make sure that those values will not be truth-y (usually, in Ruby, they are).

`{'data[alerts][follow]': False}` is not valid JSON. `False` is a syntax error (and so is single-quoted String). In my opinion, it should return an error.

**Comment by nightpool** [comment]
My mistake—I don't think this is about JSON, this is about query
parameters, which are kind of weird and unspecified

**Comment by trwnh** [comment]
Yeah, it's 100% about params. The workaround is to send form data or json instead, or e.g. to use 0 instead of False, which are not really obvious since the official expectation is that GET uses query and everything else uses formdata/json (but all 3 are "equally" understood)
